From d16997be5a2b2806d6d95b397f41cb75ad2ce85f Mon Sep 17 00:00:00 2001
From: Song Bing <b06498@freescale.com>
Date: Tue, 15 Sep 2015 14:19:02 +0800
Subject: [PATCH 0741/1691] MLK-11542 media: mxc vout: reset video output count
 to zero when streamon

commit d16997be5a2b2806d6d95b397f41cb75ad2ce85f from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

As streamoff/streamon will reset all buffer list, video output count should
be reset to zero.  Or timer will get wrong schedule time if input buffer time
stamp is 0.

Signed-off-by: Song Bing b06498@freescale.com
---
 drivers/media/platform/mxc/output/mxc_vout.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/platform/mxc/output/mxc_vout.c b/drivers/media/platform/mxc/output/mxc_vout.c
index 14f74cd..adeb6d6 100644
--- a/drivers/media/platform/mxc/output/mxc_vout.c
+++ b/drivers/media/platform/mxc/output/mxc_vout.c
@@ -2075,6 +2075,8 @@ static int mxc_vidioc_streamon(struct file *file, void *fh,
 	hrtimer_init(&vout->timer, CLOCK_REALTIME, HRTIMER_MODE_ABS);
 	vout->timer.function = mxc_vout_timer_handler;
 	vout->timer_stop = true;
+	vout->frame_count = 0;
+	vout->vdi_frame_cnt = 0;
 
 	vout->start_ktime = hrtimer_cb_get_time(&vout->timer);
 
-- 
1.9.1

